java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -1044,6 +1080,136 @@ public Table aggregate(Aggregate... aggregates) {
         aggregate.close();
       }
     }
+
+    public Table aggregateWindows(WindowAggregate... windowAggregates) {
+      // To improve performance and memory we want to remove duplicate operations
+      // and also group the operations by column so hopefully cudf can do multiple aggregations
+      // in a single pass.
+
+      // Use a tree map to make debugging simpler (columns are all in the same order)
+      TreeMap<Integer, ColumnWindowOps> groupedOps = new TreeMap<>(); // Map agg-col-id -> Agg ColOp.
+      // Total number of operations that will need to be done.
+      int totalOps = 0;
+      for (int outputIndex = 0; outputIndex < windowAggregates.length; outputIndex++) {
```

Comment:
nit: would be good to validate that `timestampColumnIndex` is not set for the window operations.

Reply:
I've added checks to bail out, if the wrong `FrameType` is used, in either window aggregation.